skia/recorder/src/main.cpp

```diff
@@ -1,217 +1,64 @@
+			                          args.maxHeight(),
+			                          args.minDuration(),
+			                          args.maxDuration(),
+			                          args.fps(),
+			                          args.bitrate());
+	}
 }
 
-#ifdef TESTING
-#else
 int main(int argc, char* argv[])
 {
-	// PARSE INPUT
-	args::ArgumentParser parser(
-	    "Record playback of a Rive file as a movie, gif, etc (eventually "
-	    "should support image sequences saved in a zip/archive too).",
-	    "Experimental....");
-	args::HelpFlag help(
-	    parser, "help", "Display this help menu", {'h', "help"});
-	args::Group required(
-	    parser, "required arguments:", args::Group::Validators::All);
-	args::Group optional(
-	    parser, "optional arguments:", args::Group::Validators::DontCare);
-
-	args::ValueFlag<std::string> source(
-	    required, "path", "source filename", {'s', "source"});
-	args::ValueFlag<std::string> destination(
-	    required, "path", "destination filename", {'d', "destination"});
-	args::ValueFlag<std::string> animationOption(
-	    optional,
-	    "name",
-	    "animation to be played, determines the numbers of frames recorded",
-	    {'a', "animation"});
-	args::ValueFlag<std::string> artboardOption(
-	    optional, "name", "artboard to draw from", {'t', "artboard"});
-	args::ValueFlag<std::string> watermarkOption(
-	    optional, "path", "watermark filename", {'w', "watermark"});
-
-	args::ValueFlag<int> width(
-	    optional, "number", "container width", {'W', "width"}, 0);
-
-	args::ValueFlag<int> height(
-	    optional, "number", "container height", {'H', "height"}, 0);
-
-	args::ValueFlag<int> small_extent_target(
-	    optional,
-	    "number",
-	    "target size for smaller edge of container",
-	    {"small-extent-target"},
-	    0);
-
-	args::ValueFlag<int> max_width(
-	    optional, "number", "maximum container width", {"max-width"}, 0);
-
-	args::ValueFlag<int> max_height(
-	    optional, "number", "maximum container height", {"max-height"}, 0);
-
-	args::ValueFlag<int> min_duration(
-	    optional, "number", "minimum duration in seconds", {"min-duration"}, 0);
-
-	args::ValueFlag<int> max_duration(
-	    optional, "number", "maximum duration in seconds", {"max-duration"}, 0);
-
-	args::ValueFlag<int> num_loops(
-	    optional,
-	    "number",
-	    "number of times this video should be looped for",
-	    {"num-loops"},
-	    1);
-
-	// 0 will use VBR. By setting a bitrate value, users enforce CBR.
-	args::ValueFlag<int> bitrate(
-	    optional, "number", "bitrate in kbps", {"bitrate"}, 0);
-
-	args::ValueFlag<std::string> snapshot_path(
-	    optional, "path", "destination image filename", {"snapshot-path"});
-
-	args::ValueFlag<float> fps(
-	    required, "number", "frame rate", {"f", "fps"}, 60.0);
-
-	args::CompletionFlag completion(parser, {"complete"});
-
 	try
 	{
-		parser.ParseCLI(argc, argv);
+		RecorderArguments args(argc, argv);
+		auto extractor = makeExtractor(args);
```

Comment:
Do we need to make sure we delete the extractor in the catches below?

Reply:
Similarly (yet minor as I think app exits), we should also delete it when we're done with it (if we're not, I see that it got removed on the left).